hudi-common/src/main/java/org/apache/hudi/BaseHoodieTableFileIndex.java

```diff
@@ -138,7 +143,20 @@ public BaseHoodieTableFileIndex(HoodieEngineContext engineContext,
     this.engineContext = engineContext;
     this.fileStatusCache = fileStatusCache;
 
-    doRefresh();
+    /**
+     * The `shouldRefresh` variable controls how we initialize the TableFileIndex:
```

Comment:
I removed `isAllInputFileSlicesCached ` and have following logic to check is all file slices cached:

```
if (cachedAllPartitionPaths == null) {
      return false;
    }
return cachedAllPartitionPaths.stream().allMatch(p -> cachedAllInputFileSlices.containsKey(p));
```

Basically, we check if all partitions are loaded. Then we check if all partitions is contained in the `cachedAllInputFileSlices`. It should be cleaner instead of maintaining a separate flag variable.